File: frontend-web/src/data/types/ValidationStandardError.ts
import { ErrorFieldMessage } from "./ErrorFieldMessage"
import { StandardError } from "./StandardError"

export class ValidationStandardError extends StandardError {
  errors: ErrorFieldMessage[]

  constructor(statusCode: number, message: string, timestamp: number, errors: ErrorFieldMessage[]) {
    super(statusCode, message, timestamp)
    this.errors = errors
  }
}

File: backend-server/src/main/kotlin/br/com/tirabassi/remotebooks/controllers/exceptions/dto/ValidationStandardError.kt
package br.com.tirabassi.remotebooks.controllers.exceptions.dto

/**
 * Classe que mapeia o erro de validação lançado pelo hibernate.
 *
 * @property statusCode O código de status HTTP.
 * @property message A mensagem de erro.
 * @property timestamp A data que ocorreu o erro.
 * @property errors A lista com os erros
 */
class ValidationStandardError(
    statusCode: Int,
    message: String,
    timestamp: Long,
    val errors: List<ErrorFieldMessage>,
) : StandardError(statusCode, message, timestamp)
